pkg/operator2/status.go

```diff
@@ -0,0 +1,50 @@
+package operator2
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+func (c *authOperator) setFailingStatus(operatorConfig *operatorv1.Authentication, reason, message string) error {
```

Comment:
Deep copy `operatorConfig` if you are going to mutate it.

Reply:
`v1helpers.UpdateStatus()` seems to be doing that for us now.